pkg/collector/autodiscovery/autoconfig.go

```diff
@@ -210,6 +206,11 @@ func (ac *AutoConfig) getAllConfigs() []check.Config {
 				errorStats.removeConfigError(cfg.Name)
 			}
 
+			// Grab any errors that occurred when reading the YAML file
+			for name, e := range fileConfPd.Errors {
```

Comment:
Not sure why this block was moved?

Reply:
The previous code block iterate over every valid config and removes all the config errors with the same config name, which is a problem because there can be multiple config with the same name. For example, if you have one valid `couch` config and one invalid `couch` config, the error will be silenced. This is the case by default with the autodiscovery (`check_name.yaml` and `check_name.d/auto_discovery.yaml`). 

Moving this block after the cleanup should fix this.